ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2

```diff
@@ -142,6 +142,26 @@ ethernet_interfaces:
 {%                     if adapter_settings.ptp.enable is arista.avd.defined %}
       enable: {{ adapter_settings.ptp.enable }}
 {%                     endif %}
+{%                     if switch.ptp["interface_config"] is arista.avd.defined %}
+{%                         if adapter_settings.ptp is arista.avd.defined %}
+{%                             if adapter_settings.ptp.enable is arista.avd.defined(true) %}
+{%                                 set ptp_config = switch.ptp["interface_config"] | arista.avd.default(adapter_settings.ptp) %}
+{%                                 if adapter_settings.ptp.endpoint_role is not arista.avd.defined('bmca') %}
+{%                                     do ptp_config.update({"role": "master"}) %}
+{%                                 endif %}
+{%                                 if adapter_settings.ptp.profile is arista.avd.defined('aes67-r16-2016') %}
+{%                                     do ptp_config.update({"announce": {"interval": 0, "timeout": 3,},"delay_req": -3,"sync_message": {"interval": -3,},"transport": "ipv4",}) %}
+{%                                 endif %}
+{%                                 if adapter_settings.ptp.profile is arista.avd.defined('smpte2059-2') %}
+{%                                     do ptp_config.update({"announce": {"interval": -1, "timeout": 3,},"delay_req": -4,"sync_message": {"interval": -4,},"transport": "ipv4",}) %}
+{%                                 endif %}
+{%                                 if adapter_settings.ptp.profile is arista.avd.defined('aes67') %}
+{%                                     do ptp_config.update({"announce": {"interval": 2, "timeout": 3,},"delay_req": 0,"sync_message": {"interval": 0,},"transport": "ipv4",}) %}
```

Comment:
Since the profiles are configurable per interface, it would be difficult to include it dynamically.

Reply:
What I meant was something like

```
{%                                 if adapter_settings.ptp.profile is arista.avd.defined('smpte2059-2') %}
{%                                     do ptp_config.update(ptp_smtpe2059_2_profile) %}
```

where `ptp_smtpe2059_2_profile` is a variable coming from default_var for instance. 
This could be reused there then: https://github.com/aristanetworks/ansible-avd/pull/1916/files/626f51239ff06f0d070db500fc1a4c305f0f254b#diff-71e34860f7c82535dbd35d9f2284206ac473b7d52604b8d9e1b6e5caf5b5e100R595

The three profiles are well known and hence quite static